mlfinlab/sample_weights/attribution.py

```diff
@@ -0,0 +1,84 @@
+"""
+Logic regarding return and time decay attribution for sample weights from chapter 4.
+"""
+
+import pandas as pd
+import numpy as np
+from mlfinlab.util.multiprocess import mp_pandas_obj
+from mlfinlab.sampling.concurrent import num_concurrent_events, get_av_uniqueness_from_tripple_barrier
+
+
+def _apply_weight_by_return(label_endtime, num_conc_events, close_series, molecule):
+    """
+    Snippet 4.10, page 69, Determination of Sample Weight by Absolute Return Attribution
+    Derives sample weights based on concurrency and return. Works on a set of
+    datetime index values (molecule). This allows the program to parallelize the processing.
+    :param label_endtime: (pd.Series) Label endtime series (t1 for triple barrier events)
+    :param num_conc_events: (pd.Series) number of concurrent labels (output from num_concurrent_events function).
+    :param close_series: (pd.Series) close prices
+    :param molecule: (an array) a set of datetime index values for processing.
+    :return: (pd.Series) of sample weights based on number return and concurrency for molecule
+    """
+
+    ret = np.log(close_series).diff()  # log-returns, so that they are additive
```

Comment:
Please start comments with a capital letter and if it is appropriate - standard sentence grammar.